gef-ui/api-tests/facility-end-date.post.api-test.js

```diff
@@ -0,0 +1,199 @@
+jest.mock('csurf', () => () => (req, res, next) => next());
+jest.mock('../server/middleware/csrf', () => ({
+  ...jest.requireActual('../server/routes/middleware/csrf'),
+  csrfToken: () => (req, res, next) => next(),
+}));
+
+const { when, resetAllWhenMocks } = require('jest-when');
+const { MAKER } = require('../server/constants/roles');
+const { withRoleValidationApiTests } = require('./common-tests/role-validation-api-tests');
+const app = require('../server/createApp');
+const { post } = require('./create-api').createApi(app);
+const api = require('../server/services/api');
+const storage = require('./test-helpers/storage/storage');
+
+api.getFacility = jest.fn();
+api.updateFacility = jest.fn();
+api.updateApplication = jest.fn();
+
+const dealId = '123';
+const facilityId = '111';
+
+describe('facility-end-date routes', () => {
+  beforeEach(async () => {
+    resetAllWhenMocks();
+    await storage.flush();
+    jest.clearAllMocks();
+  });
+
+  afterAll(async () => {
+    resetAllWhenMocks();
+    jest.resetAllMocks();
+    await storage.flush();
+  });
+
+  describe('POST /application-details/:dealId/facilities/:facilityId/facility-end-date', () => {
+    beforeEach(() => {
+      when(api.getFacility)
+        .calledWith({ facilityId, userToken: expect.anything() })
+        .mockResolvedValueOnce({
+          details: {
+            isUsingFacilityEndDate: null,
+            coverStartDate: '2024-07-15T00:00:00.000Z',
+          },
+        });
+    });
+
+    describe('with saveAndReturn false', () => {
```

Comment:
whats the reasoning for not testing this route more?

Reply:
refactored to make this more clear & some extra tests